### ISSUE TITLE
T? improperly permitted when T is constrained to an interface

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#33493]
The following snippet of code from our regression tests should complain that `T?` is not permitted because it is not known to be a value type or a reference type.  But we permit it without error.

``` c#
    static void F5<T>(T t5) where T : I<T?>
```

**Comment by gafter** [comment]
This appears to have been fixed.
